modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/sorted/PageMemorySortedIndexStorage.java

```diff
@@ -120,26 +119,60 @@ public void remove(IndexRow row) {
 
     @Override
     public Cursor<IndexRow> scan(@Nullable BinaryTuplePrefix lowerBound, @Nullable BinaryTuplePrefix upperBound, int flags) {
-        Cursor<SortedIndexRow> cursor;
+        boolean includeLower = (flags & GREATER_OR_EQUAL) != 0;
+        boolean includeUpper = (flags & LESS_OR_EQUAL) != 0;
+
+        SortedIndexRowKey lower = createLowerBound(lowerBound, includeLower);
+
+        SortedIndexRowKey upper = createUpperBound(upperBound, includeUpper);
 
         try {
-            cursor = sortedIndexTree.find(
-                    toSortedIndexRowKey(lowerBound),
-                    toSortedIndexRowKey(upperBound),
-                    (flags & GREATER_OR_EQUAL) != 0,
-                    (flags & LESS_OR_EQUAL) != 0,
-                    null,
-                    null
-            );
+            Cursor<SortedIndexRow> cursor = sortedIndexTree.find(lower, upper, includeLower, includeUpper, null, null);
```

Comment:
Passing these flags is pointless I believe. Simple `sortedIndexTree.find(lower, upper)` will give us the same result